addons/account/models/account_bank_statement_line.py

```diff
@@ -661,12 +661,29 @@ def _retrieve_partner(self):
 
         # Retrieve the partner from the partner name.
         if self.partner_name:
-            domain = [
-                ('parent_id', '=', False),
-                ('name', 'ilike', self.partner_name),
-            ]
-            for extra_domain in ([('company_id', '=', self.company_id.id)], []):
-                partner = self.env['res.partner'].search(extra_domain + domain, limit=1)
+            for domain in (
+                [
+                    ('parent_id', '=', False),
+                    ('name', '=ilike', self.partner_name),
+                    ('company_id', '=', self.company_id.id),
+                ],
+                [
+                    ('parent_id', '=', False),
+                    ('name', '=ilike', self.partner_name),
+                    ('company_id', '=', False),
+                ],
+                [
+                    ('parent_id', '=', False),
+                    ('name', 'ilike', self.partner_name),
+                    ('company_id', '=', self.company_id.id),
+                ],
+                [
+                    ('parent_id', '=', False),
+                    ('name', 'ilike', self.partner_name),
+                    ('company_id', '=', False),
+                ],
```

Comment:
I don't understand why it repeats twice

Reply:
The diff is the `ilike`.
Once it's `=ilike` ---> exact matching
then it's `ilike` ---> with %stuff%